### ISSUE TITLE
update rule: react-a11y-anchors - Do not check if role='button'

### ISSUE THREAD

**Issue by ipip2005** [Microsoft/tslint-microsoft-contrib#295]
manytimes anchors are used as a button:
```
<a role='button' onclick="...">
  <span classname='...'></span>
</a>
```
They should NOT be checked with react-a11y-anchors rule or `at least 4 characters long` condition

**Comment by ipip2005** [comment]
I also found this rule fails on
`<a> <img /> </a>`
Perhaps we should remove this check of `at least 4 characters long` because variant usage of anchors will fail on this rule.

**Comment by ipip2005** [comment]
Also in product, actually it is hard to find a plain text in the link because of localization requirements.

**Comment by ipip2005** [comment]
Another update is that we should allow role='presentation' appears with non-empty alt text. Like 
                <img
                  aria-hidden={ true }
                  role='presentation'
                  src=...
                  alt=... />
In following situation:
<img alt='123'/>
<span>123</span.
Sometimes the alt text equals to the nearest text. The alt text is no need to be discoverable by assistive technologies. But they should exist to show for sighted user once this img fails loading.

**Comment by ipip2005** [comment]
As a result, all anchors, if they are not used as real `anchors`, they should have explicit role specified (mostly role='button')
For example
1. 
```
<a role='button' onclick='...'>
   <i className='some-icon-class' />
</a>
```
2.
```
<a role='button' onclick='...'>
   <img src='...' alt='...'/>
</a>
```
